### PR TITLE
Ft notification

### PR DESCRIPTION
**Pull request by PatrickTUy**
# PR Description
Creation of notifications
# Description of tasks that were expected to be completed

1.  users are able to see notifications
2. users are able to mark notification as read 
3. users are able to delete notification
4. responsiveness

# functionality 

- use this [link](https://devpulse-review-app-40.herokuapp.com/) for review app
- sign in with an email e.g(admin@random.com), use any characters of your choice as password 
- click on the notification icon on top right, click on the text of notification to mark it as read
- click on the cross icon to delete it, finally click on the mark all as read button to mark all of them as read.

# How has this been tested?
We created test that implemented the functionality of all mouse clicks and key downs to run all  the functions of reading and deleting.
in the terminal, type npm run test && npm run coverage

# Screenshots (If appropriate)
Screen Shot 2022-07-22 at 10 56 57
Screen Shot 2022-07-22 at 10 54 39

# Please check this Checklist before you submit your PR:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My code generate no warnings
- [x] My test coverage meet the set test coverage threshold
- [x] There are no vulnerabilities
- [x] There are no conflicts with the base branch

### CODE REVIEW

**Comment by feyton** [comment]
I deployed a review app for you to try out this pull request 👇🏽

Heroku App: https://devpulse-review-app-40.herokuapp.com/


<hidden purpose="for-rewritable-pr-comment-action-use" value="Deploy"></hidden>

**Review by chanceDira — APPROVED** [review]
This looks good 👍

File: package.json [hunk 1]
```diff
@@ -10,8 +10,8 @@
   "scripts": {
     "dev": "webpack server",
     "build": "webpack",
-    "test": "npx jest",
-    "coverage": "jest --coverage --silent",
+    "test": "npx jest -u",
```
Comment (feyton) [inline package.json:13]: This command should not change as the purpose of the tests is to check if other files were affected. If you just run the command with `-u`, then it serves no purpose.

File: package.json [hunk 2]
```diff
@@ -10,8 +10,8 @@
...
+    "coverage": "jest --coverage --silent -u",
```
Comment (feyton) [inline package.json:14]: ### Critical 

Same here, restore the previous command. You have to be sure to update snapshot before you push the code not when running the github actions

File: src/components/Notification.tsx [hunk 3]
```diff
@@ -0,0 +1,217 @@
+import React, { useState } from 'react';
+import { XIcon } from '@heroicons/react/solid';
+import { format } from 'date-fns';
+import { useTranslation } from 'react-i18next';
+
+const notificationDummy = [
```
Comment (feyton) [inline src/components/Notification.tsx:6]: Can we move those data outside the function into its own file

File: src/components/Notification.tsx [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+                  <div
+                    className="flex flex-col w-full gap-[5px] cursor-pointer"
+                    onClick={() => {
+                      markRead(
+                        notification.id,
+                        notifications,
+                        setNotifications,
+                      );
+                    }}
+                    data-testid={index === 0 && 'read'}
```
Comment (feyton) [inline src/components/Notification.tsx:159]: don't use a variable data-test-id as they will break your code. You can safely mock the data you are expecting same case with an `api` call

File: src/components/Notification.tsx [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+                  <div className="flex flex-col items-center transition-all">
+                    <div
+                      className={`h-[10px] w-[10px] rounded-full ${
+                        !notification.read
+                          ? 'bg-[#148FB6]'
+                          : 'border-[rgba(95,91,91,0.5)] dark:border-white border-[1px]'
+                      }  mt-[7px] mb-[10px]`}
+                    />
+                    <XIcon
+                      className="border-[rgba(95,91,91,0.5)] dark:fill-white h-[20px] w-[20px] cursor-pointer"
```
Comment (feyton) [inline src/components/Notification.tsx:181]: Avoid absolute values in the styling, instead, put the color in our `tailwind.css.config` to make it easier to replace the colors when the time comes

**Review by feyton — CHANGES_REQUESTED** [review]
### Critical changes

Please refer to the comments I left especially in the package.json file

**Review by Dev-nkundineza — APPROVED** [review]
Great job @PatrickTUy and @Tsa2341

**Review by jeandamascyifuzo — APPROVED** [review]
Geat Job

**Review by samyshyaka — CHANGES_REQUESTED** [review]
- I think it's better if the background is not blurred when the drop-down appears
- Also, you can try to move the drop-down closer to the header. I think it will be even better if it touches the header
- Lastly, try to shift the drop-down a few pixels to the right to make sure that it is right under the notification icon

**Comment by PatrickTUy** [comment]
> * I think it's better if the background is not blurred when the drop-down appears
> * Also, you can try to move the drop-down closer to the header. I think it will be even better if it touches the header
> * Lastly, try to shift the drop-down a few pixels to the right to make sure that it is right under the notification icon

We had actually implemented what was in figma  which was approved [ link ](https://www.figma.com/proto/Q4kozKNucJX4kJsHgnB4Vi/ATLP-UI-V2?node-id=1699%3A24308&scaling=contain&page-id=58%3A97&starting-point-node-id=216%3A1284)

**Review by feyton — APPROVED** [review]

**Review by ceelogre — COMMENTED** [review]
There are no instructions on how to test this.

**Review by ceelogre — DISMISSED** [review]
1. Let's use black people for profile images
2. On mobile devices, the notification container occupies the whole screen

IMG_D2BB256E07D7-1
.

**Review by ceelogre — APPROVED** [review]

**Pull request merged**